lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -429,18 +472,15 @@ assignMemoryLayouts(llvm::SmallVector<std::tuple<Operation *, int, Operation *>>
         continue;
     }
 
+    bool loadIsMMAV3 = false, hasSharedEncoding = false;
     if (use->hasTrait<OpTrait::DotLike>()) {
-      loadInfo.usedByDot = true;
       if (loadIsMMAv3(op)) {
-        loadInfo.loadIsMMAV3 = true;
-        loadInfo.sharedEncoding =
-            getSharedEncoding(op, /*loadIsMMAv3=*/true).value_or(nullptr);
+        loadIsMMAV3 = true;
```

Comment:
I feel like this logic could be simplified now - for example does it make sense to keep track of `loadIsMMAV3`? `loadIsMMAV3` is true iff `hasSharedEncoding` is also true, it seems.